### DISCUSSION TITLE
.NET 9 - Release

### DISCUSSION THREAD

**Discussion by jamesmontemagno** [dotnet/core#9597] [News]
We are pleased to announce the general availability of .NET 9. 

Read about the final release on the [.NET Blog announcement](https://aka.ms/dotnet9). 

You can find a detailed overview of new features across .NET 9 in our documentation:

- [What's new in .NET 9](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/overview): [Runtime](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/runtime), [Libraries](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/libraries), and [SDK](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/sdk)
- [What's new in C# 13](https://learn.microsoft.com/dotnet/csharp/whats-new/csharp-13)
- [What's new in F# 9](https://learn.microsoft.com/dotnet/fsharp/whats-new/fsharp-9)
- [What's new in ASP.NET Core](https://learn.microsoft.com/aspnet/core/release-notes/aspnetcore-9.0)
- [What's new in .NET Aspire](https://learn.microsoft.com/dotnet/aspire/whats-new/)
- [What's new in .NET MAUI](https://learn.microsoft.com/dotnet/maui/whats-new/dotnet-9)
- [What's new in EF Core](https://learn.microsoft.com/ef/core/what-is-new/ef-core-9.0/whatsnew)
- [What's new in WPF](https://learn.microsoft.com/dotnet/desktop/wpf/whats-new/net90)
- [What's new in Windows Forms](https://learn.microsoft.com/dotnet/desktop/winforms/whats-new/net90)

## Get started

To get started with .NET 9, [install the .NET 9 SDK](https://dotnet.microsoft.com/download/dotnet/9.0).

If you're on Windows using Visual Studio, we recommend installing the latest [Visual Studio 2022](https://visualstudio.microsoft.com/download/). .NET 9 is now can be installed directly through the Visual Studio installer starting with Visual Studio 2022 17.12.

You can also use Visual Studio Code and the [C# Dev Kit](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit) extension with .NET 9.

You can find [.NET 9 Installer & Binaries](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/9.0.0.md) on GitHub.

## Release Notes

Release notes are available for changes between .NET 9 RC2 and this release:

* [Libraries](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/libraries.md)
* [Runtime](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/runtime.md)
* [SDK](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/sdk.md)
* [.NET MAUI](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/dotnetmaui.md)
* [ASP.NET Core](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/aspnetcore.md)
* [C#](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/csharp.md)
* [EF Core and Data](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/efcoreanddata.md)
* [F#](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/fsharp.md)
* [WinForms](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/winforms.md)
* [WPF](https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.0/wpf.md)

Celebrate the release of .NET 9 at [.NET Conf 2024](https://dotnetconf.net) - November 12th -14th.

**Comment by Atulin** [comment]
Very available indeed
![image](https://github.com/user-attachments/assets/ea1bda03-ee01-44c4-9c22-37d38064c24c)
![image](https://github.com/user-attachments/assets/e178d2f2-c5f9-425e-97aa-28f0daeb8c30)

**Reply by GerardSmit** [reply]
This depends on https://github.com/microsoft/winget-pkgs/pull/190763. You can subscribe to this PR to see when it's available in WinGet.

**Reply by Atulin** [reply]
Sure, of course. I just find it weird that Microsofts builds the hype around a release first with "HOLY SHIT GUYS .NET 9 IS AVAILABLE TODAY DOWNLOAD IT NOW IT'S NOW HOOOOLY DOWNLOAD IT RIGHT THIS INSTANT" kinds of blogposts, articles, and discussions... only for the download page and Winget to show only RC2.

The download page now, thankfully, has the 9.0.100 properly, but Winget is still missing... and yet the general messaging is "GET IT NOW"

Just maybe, you know, fully release things *first*, then write the articles

**Reply by jamesmontemagno** [reply]
Thanks for the comment and feedback here. Will work with the team and figure out what is going on here, it is a challenge with many moving parts as to when to make the post as bits are available, but will work to make it much more granular.

**Reply by jamesmontemagno** [reply]
@Atulin winget packages should now be available

**Comment by craigs100** [comment]
Thanks but it's not on "General" release though is it?  Let me explain.....

I develop Umbraco websites using the .Net Core SDK's on Linux Mint.  Following your (Microsoft's) instructions not to mix repos between MS & the distro's repos, I opted to go with the MS repos.  I generally need the SDK's asap, plus I want to be sure I'm using an approved SDK version as I don't want any surprises when I deploy code to servers.

From .NET V9, you're now not providing the Linux versions for download if the Linux Distro of choice provides a version in their repo.

This now gives me a massive problem. I need versions that aren't available on the Linux Mint repo which includes V6 and V9.

Could you please reconsider this?

The distro's aren't particularly quick making new SDK's available. Also, do you know if the versions they do provide are 1 to 1 compatible with yours.

Thanks for reading so far. I'm extremely keen for answers as it's my livelihood being affected.

Regards.

**Comment by bettwedder** [comment]
With .NET 9 I started getting System.AccessViolationException crashes in a .NET MAUI application that was rock solid before the upgrade.  The issue is inside a function that reads and sums an array in parallel.  The program runs the function that throws this error twice each millisecond and will run for anywhere from 25 minutes to 2 hours without it encountering the error.   An example message is at the bottom of this post.  While writing this post I got a new crash reason:

Application: MyMauiApp.exe
CoreCLR Version: 9.0.24.52809
.NET Version: 9.0.0
Description: The process was terminated due to an internal error in the .NET Runtime at IP 0x00007FFE80BBEA8B (0x00007FFE808D0000) with exit code 0xc0000096.


Previous Crash Errors:

CoreCLR Version: 9.0.24.52809
.NET Version: 9.0.0
Description: The process was terminated due to an unhandled exception.
Exception Info: System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.

Stack:
   at System.Runtime.ExceptionServices.InternalCalls.<RhpSfiInit>g____PInvoke|0_0(System.Runtime.StackFrameIterator*, Void*, Int32, Boolean*)
   at System.Runtime.ExceptionServices.InternalCalls.<RhpSfiInit>g____PInvoke|0_0(System.Runtime.StackFrameIterator*, Void*, Int32, Boolean*)
   at System.Runtime.EH.DispatchEx(System.Runtime.StackFrameIterator ByRef, ExInfo ByRef)
   at System.Runtime.EH.RhThrowHwEx(UInt32, ExInfo ByRef)
   at MyNamespace.MyClass+<>c__DisplayClass96_0.<CalculateMomemtumValues>b__0(Int32)
   at System.Threading.Tasks.Parallel+<>c__DisplayClass19_0`2[[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Int32, System.Private.CoreLib, Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<ForWorker>b__1(System.Threading.Tasks.RangeWorker ByRef, Int64, Boolean ByRef)
   at System.Threading.Tasks.TaskReplicator+Replica.Execute()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(System.Threading.Thread, System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef, System.Threading.Thread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()

**Reply by julealgon** [reply]
You should consider opening an issue on the .NET MAUI repo.

**Reply by bettwedder** [reply]
> You should consider opening an issue on the .NET MAUI repo.

You are correct, I just opened it in the runtime repo as the error seems to be non-MAUI related.   I can't get it to happen in debug mode which leads me to believe it has something to do with runtime optimizations.  Should I delete the one here?

**Reply by julealgon** [reply]
> Should I delete the one here?

Probably not needed. I would link your runtime issue here though for tracking.

**Comment by pilarodriguez** [comment]
.Net 9 is not yet available in all regions. At least it is not available in China North
![image](https://github.com/user-attachments/assets/772a1415-04ca-485a-8d0e-710d77d04a4d)

I get the following error when I deploy a .net 9 app service
```
You must install or update .NET to run this application.

App: C:\home\site\wwwroot\Nfield.Systems.Monitoring.dll
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '9.0.0' (x64)
.NET location: C:\Program Files\dotnet\

The following frameworks were found:
  2.2.14 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  3.0.3 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  3.1.32 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  5.0.17 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  6.0.32 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  7.0.20 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  8.0.8 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  8.0.10 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  9.0.0-rc.1.24431.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

Learn more:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&amp;framework_version=9.0.0&amp;arch=x64&amp;rid=win-x64&amp;os=win10
```
When is it going to be rolled out to all regions?

**Reply by mairaw** [reply]
Where is that screenshot from? .NET 9 is available globally so this might be related to a particular service.

**Reply by pilarodriguez** [reply]
It is in the Azure portal, the Configuration blade of the app service

![image](https://github.com/user-attachments/assets/aac32d8b-12f1-4537-82d8-cac29ba796fc)

It is a regular app service

I can't select .net 9 in the portal. And if I deploy my .net 9 app service using bicep, the app doesn't start and I get the error I mentioned in the log file.

```
HTTP Error 500.31 - ANCM Failed to Find Native Dependencies
Common solutions to this issue:
The specified version of Microsoft.NetCore.App or Microsoft.AspNetCore.App was not found.
Troubleshooting steps:
Check the system event log for error messages
Enable logging the application process' stdout messages
Attach a debugger to the application process and inspect
For more information visit: https://go.microsoft.com/fwlink/?LinkID=2028526
```
```
You must install or update .NET to run this application.

App: C:\home\site\wwwroot\Nfield.Systems.Monitoring.dll
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '9.0.0' (x64)
.NET location: C:\Program Files\dotnet\

The following frameworks were found:
  2.2.14 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  3.0.3 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  3.1.32 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  5.0.17 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  6.0.32 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  7.0.20 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  8.0.8 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  8.0.10 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  9.0.0-rc.1.24431.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

Learn more:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&amp;framework_version=9.0.0&amp;arch=x64&amp;rid=win-x64&amp;os=win10
```

It works fine if I deploy to a non China region, or if I use .net 8 instead

**Comment by craigs100** [comment]
Still no answer on publishing SDK's for Linux as they used to.  Linux Mint still only has SDK 8 in it repos!  This is no good for professional devs.
